This PR can be renamed ; Frontend

Kaxuo opened the pull request.

jeremykairis commented on FrontEnd/src/index.html:
```diff
@@ -1,13 +1,52 @@
+  </head>
+  <body class="body">
+    <app-root></app-root>
+    <!-- JQuery -->
```
You should avoid jQuery in Angular apps and use npm to add dependencies.
I checked mdbbootstrap and found their pricing annoying so it's better to learn [Angular Material](https://material.angular.io/) and [Bootstrap](https://getbootstrap.com/).

jeremykairis commented on FrontEnd/src/app/components/profile-page/profile-page.component.ts:
```diff
@@ -0,0 +1,146 @@
+          if (values.number) {
+            this.data.number = values.number;
+            this.numberEdit = false;
+          }
```
Could you simplify this code being more generic ?

jeremykairis commented on FrontEnd/src/app/components/profile-page/profile-page.component.ts:
```diff
@@ -0,0 +1,146 @@
+            this.hobby = new FormGroup({
+              hobby: new FormControl(this.data.hobby, [Validators.required]),
+            });
+          });
```
Do you really need all these flags and separate FormGroups ?

jeremykairis commented on FrontEnd/src/app/components/navbar/navbar.component.scss:
```diff
@@ -0,0 +1,20 @@
+
```
Remove unused code.

jeremykairis commented on BackEnd/Profile/AutoMapperProfile.cs:
```diff
@@ -0,0 +1,15 @@
+    {
+        public AutoMapperProfile()
+        {
+            CreateMap<User, UserModel>();
+            CreateMap<RegisterModel, User>();
+            CreateMap<UpdateModel, User>();
```
Try to simplify the user update process to reduce mappers.

jeremykairis commented on BackEnd/Migrations/UserContextModelSnapshot.cs:
```diff
@@ -0,0 +1,60 @@
+    {
+        protected override void BuildModel(ModelBuilder modelBuilder)
+        {
+#pragma warning disable 612, 618
```
Why do you use this #pragma ?

jeremykairis commented on BackEnd/appsettings.json:
```diff
@@ -1,4 +1,10 @@
 {
+  "ConnectionStrings": {
+    "DefaultConnection": "server=localhost;port=3306;database=csharp;user=root;password=Da4EhBEQ5FiUmleVeFF1;CharSet=utf8"
+  },
+  "AppSettings": {
+    "Secret": "THIS IS USED TO SIGN AND VERIFY JWT TOKENS, REPLACE IT WITH YOUR OWN SECRET, IT CAN BE ANY STRING"
+  },
```
Sensitive data should be kept out of the source control.

Jeykairis reviewed (changes requested):
Please, take the time to cleanup your code before submitting a PR. 
Rename the branch and the PR as it contains code from the backend. 
Use a proper .gitignore for the backend.
Remove extra white spaces and ensure clarity and consistency of variable names to be able to remove most comments.
Finally, try to keep the code as simple as possible.

Kaxuo commented:
Everything should have been changed like you asked (except the name of the branch, I will try to be more accurate next time)

- Codes for the Profile Page has been totally changed, flags are added differently (and thus, not seen) , the "sendData" method doesn't have a "if" for each field anymore and is more generic. The FormGroup is more consice, no more FormGroup for each input field. ( commit : "Changes done to the frontend" )
Sorry, I did that at first because I couldn't find a solution for a "Single Field only Input" so I just coded it the hard way.

- Added some lines in the gitignore file to ignore what needs to be ignored in the subfolder "BackEnd"
(commit : "Remove ignored files")

- Sensitive Data are now stored safely and can no longer be seen in the BackEnd , appsetting.json
(commit : "ChangesRequested Done")

- Deleted the UpdateModel because it wasn't needed anymore. 
(commit : "CleanUp")

- #Pragma file is automatically generated when migrating and updating the database.

I hope I did everything correctly. Sorry if I took some time to do it. I got stuck trying to find a way to hide sensitive data when the solution was so simple.

Many Thanks

jeremykairis commented on BackEnd/Controllers/UserController.cs:
```diff
@@ -0,0 +1,170 @@
+            var model = _mapper.Map<IList<UserModel>>(users);
+            return Ok(model);
+        }
+
+
```
Be consistent using white spaces.

jeremykairis commented on BackEnd/Data/MySQLMethods.cs:
```diff
@@ -0,0 +1,179 @@
+        {
+            return _context.Users.Find(id);
+        }
+
```
Take care of white spaces to make clean code.

jeremykairis commented on BackEnd/Data/UserContext.cs:
```diff
@@ -0,0 +1,41 @@
+        {
+        }
+
+        // protected override void OnModelCreating(ModelBuilder modelBuilder)
```
Remove the old code.

jeremykairis commented on FrontEnd/src/index.html:
```diff
@@ -1,13 +1,32 @@
+    <base href="/" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="icon" type="image/x-icon" href="favicon.ico" />
+    <link
```
Can you use npm to add these links ?

Jeykairis reviewed (changes requested):
It gets better and better, continue to improve the .gitignore and cleanup white spaces to get it merged.

Kaxuo commented:
- White spaces have been cleanup a little bit better ( hopefully ) and made more consistent
- Sorry about unused codes, most of the times, they're here for me as a reminder that either I need to check up on that later or that it "exists" so that I could use it again somewhere else.
- Links has been removed in the index.html, they're not needed. I probably added them when trying to make MDB work when a simple npm install would have sufficed
- I probably made a mistake, I did a git push -f origin and it closed the pull request. It says that It has a different history now, I can't find a way to re open it. I messed around and in the end , I just went ahead and forced push to the master.
I will be more careful next time for the next pull request. Sorry for the inconvenience, Thanks
